### PR TITLE
KAFKA-16908: Refactor `QuorumConfig` with `AbstractConfig`

### PR DESCRIPTION
**Pull request by xijiu**
As title

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
  * controller should be able to transition from standby to active without reloading all of
  * the metadata. The standby is a "hot" standby, not a "cold" one.
  */
-public class QuorumConfig {
+public class QuorumConfig extends AbstractConfig {
```
Comment (chia7712) [inline raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java]: Does it need to extend `AbstractConfig` if it's constructor accepts a `AbstractConfig` already?

**Review by chia7712 — COMMENTED** [review]
@xijiu thanks for this patch!

File: raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java (hunk 1 above)
Comment (xijiu) [reply raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java]: > Does it need to extend `AbstractConfig` if it's constructor accepts a `AbstractConfig` already?

Indeed, having it extend AbstractConfig is unnecessary. And I have removed it.   PTAL

File: raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java [hunk 2]
```diff
@@ -109,62 +109,34 @@ public class QuorumConfig {
             .define(QUORUM_REQUEST_TIMEOUT_MS_CONFIG, INT, DEFAULT_QUORUM_REQUEST_TIMEOUT_MS, null, MEDIUM, QUORUM_REQUEST_TIMEOUT_MS_DOC)
             .define(QUORUM_RETRY_BACKOFF_MS_CONFIG, INT, DEFAULT_QUORUM_RETRY_BACKOFF_MS, null, LOW, QUORUM_RETRY_BACKOFF_MS_DOC);
 
-    private final int requestTimeoutMs;
```
Comment (chia7712) [inline raft/src/main/java/org/apache/kafka/raft/QuorumConfig.java:112]: Could you please keep those local variables since they are not dynamic?

**Review by chia7712 — COMMENTED** [review]
@xijiu thanks for this patch. Could you please add a `QuorumConfig` reference to `KafkaConfig` and then remove the unused quorum variables from `KafkaConfig`? `quorumElectionTimeoutMs`, `quorumFetchTimeoutMs`, `quorumElectionBackoffMs`, `quorumLingerMs`, and `quorumRetryBackoffMs`?

The above behavior is similar to `remoteLogManagerConfig` in `KafkaConfig`

**Comment by xijiu** [comment]
@chia7712    I have modified the code  and remove some getter methods from `KafkaConfig`,    PTAL

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 3]
```diff
@@ -231,6 +231,9 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
   private val _remoteLogManagerConfig = new RemoteLogManagerConfig(this)
   def remoteLogManagerConfig = _remoteLogManagerConfig
 
+  private val _quorumConfig = new QuorumConfig(this)
+  def quorumConfig = _quorumConfig
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: nit: please declare the type explicitly

**Review by chia7712 — APPROVED** [review]
LGTM

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 3 above)
Comment (xijiu) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: > nit: please declare the type explicitly

Done, I have declared the `QuorumConfig` type, PTAL

**Pull request merged**